src/main/java/rx/internal/operators/OperatorStringSplit.java

```diff
@@ -0,0 +1,156 @@
+package rx.internal.operators;
+
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.regex.Pattern;
+
+import rx.Observable.Operator;
+import rx.Producer;
+import rx.Subscriber;
+import rx.functions.Action0;
+import rx.subscriptions.Subscriptions;
+
+public class OperatorStringSplit implements Operator<String, String> {
+    private final Pattern pattern;
+
+    public OperatorStringSplit(final Pattern pattern) {
+        this.pattern = pattern;
+    }
+
+    @Override
+    public Subscriber<? super String> call(Subscriber<? super String> child) {
+        StringSplitSubscriber cs = new StringSplitSubscriber(child, pattern);
+        child.add(cs);
+        StringSplitProducer cp = new StringSplitProducer(cs);
+        child.setProducer(cp);
+        return cs;
+    }
+
+    static final class StringSplitProducer implements Producer {
+        private StringSplitSubscriber cs;
```

Comment:
literally copied pasted the code from OperatorConcat and started changing it to string split. Many of the inner classes in Rx are final. Is there a problem?